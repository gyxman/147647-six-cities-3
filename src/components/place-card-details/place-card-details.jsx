import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

// const PlacesListWrapped = withActiveItem(PlacesList);

export class PlaceCardDetails extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      activeOffer: null,
    };
  }

  _offerTitleClickHandler(offer) {
    this.props.onOfferTitleClick(offer);
  }

  _offerTitleHoverHandler(offer) {
    this.setState({activeOffer: offer});
  }

  render() {
    const {offer} = this.props;
    // const {activeOffer} = this.state;
    const {
      title,
      description,
      images,
      price,
      type,
      isFavorite,
      rating,
      bedrooms,
      adults,
      goods,
      host,
    } = offer;
    const {name, avatarUrl, isPro} = host;
    const fixRating = (Math.round(rating) / 5) * 100;

    return (
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {images.map((item, i) => (
                <div key={`photo-${i}`} className="property__image-wrapper">
                  <img className="property__image" src={item} alt={title}/>
                </div>
              ))}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {isFavorite && (
                <div className="property__mark">
                  <span>Premium</span>
                </div>
              )}
              <div className="property__name-wrapper">
                <h1 className="property__name">{title}</h1>
                <button className="property__bookmark-button button" type="button">
                  <svg className="property__bookmark-icon" width="31" height="33">
                    <use xlinkHref="#icon-bookmark"/>
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: `${fixRating}%`}}/>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{rating}</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">{type}</li>
                <li className="property__feature property__feature--bedrooms">
                  {bedrooms}
                </li>
                <li className="property__feature property__feature--adults">
                  {adults}
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {goods.map((item, i) => (
                    <li key={`equipment-${i}`} className="property__inside-item">
                      {item}
                    </li>
                  ))}
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div
                    className={`property__avatar-wrapper user__avatar-wrapper ${
                      isPro ? `property__avatar-wrapper--pro` : ``
                    }`}
                  >
                    <img
                      className="property__avatar user__avatar"
                      src={avatarUrl}
                      width="74"
                      height="74"
                      alt={name}
                    />
                  </div>
                  <span className="property__user-name">{name}</span>
                </div>
                <div className="property__description">
                  {description}
                </div>
              </div>
              {/* <ReviewsList reviews={reviews} /> */}
            </div>
          </div>
          {/* TODO вывести карту и туда предложения рядом */}
          {/* <Map */}
          {/*  className={`property__map`}*/}
          {/*  offers={neighbourhoods.map((neighbourhood) => neighbourhood.coords)}*/}
          {/*  activeOffer={activeOffer ? activeOffer.coords : activeOffer}*/}
          {/* /> */}
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            {/* <PlacesListWrapped className={`near-places__`} */}
            {/*  isTabs={false}*/}
            {/*  offers={neighbourhoods}*/}
            {/*  onHover={this._offerTitleHoverHandler.bind(this)}*/}
            {/*  onSelect={this._offerTitleClickHandler.bind(this)} />*/}
          </section>
        </div>
      </main>
    );
  }
}

PlaceCardDetails.propTypes = {
  offer: PropTypes.shape({
    title: PropTypes.string.isRequired,
    description: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    images: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    price: PropTypes.number.isRequired,
    time: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    isPremium: PropTypes.bool.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    rating: PropTypes.number.isRequired,
    bedrooms: PropTypes.string.isRequired,
    adults: PropTypes.string.isRequired,
    goods: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    host: PropTypes.shape({
      avatarUrl: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      isPro: PropTypes.bool.isRequired,
    }).isRequired,
    coords: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired,
    reviews: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string.isRequired,
          image: PropTypes.string.isRequired,
          rating: PropTypes.number.isRequired,
          description: PropTypes.string.isRequired,
          date: PropTypes.string.isRequired,
        })).isRequired,
  }).isRequired,
  onOfferTitleClick: PropTypes.func.isRequired,
};
