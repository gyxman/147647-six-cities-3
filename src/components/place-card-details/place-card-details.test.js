import React from "react";
import renderer from "react-test-renderer";
import {PlaceCardDetails} from "./place-card-details.jsx";
import {PlaceType} from "../../enums/place-type.enum";

const mock = {
  name: `Beautiful & luxurious apartment at great location`,
  description: [
    `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
  ],
  image: `img`,
  photos: [
    `img2`,
    `img3`
  ],
  price: 120,
  time: `night`,
  type: `Apartment`,
  isPremium: true,
  rating: 3.1,
  countOfBedrooms: `3 Bedrooms`,
  maxCountOfGuests: `Max 4 adults`,
  equipment: [
    `Wifi`, `Heating`, `Kitchen`, `Cable TV`
  ],
  owner: {
    picture: `img4`,
    name: `Mike`,
    isSuper: true,
  },
  coords: [52.3909553943508, 4.929309666406198],
  reviews: [
    {
      name: `Max`,
      image: `https://picsum.photos/54/54`,
      rating: 4.1,
      description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      date: `2019-04-24`
    }
  ]
};

const mockNeighbourhoods = [
  {
    name: `Beautiful & luxurious apartment at great location`,
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    image: `https://picsum.photos/260/200`,
    photos: [
      `https://picsum.photos/260/200`,
      `https://picsum.photos/260/200`,
      `https://picsum.photos/260/200`,
      `https://picsum.photos/260/200`,
      `https://picsum.photos/260/200`,
      `https://picsum.photos/260/200`
    ],
    price: 120,
    time: `night`,
    type: PlaceType.Apartment,
    isPremium: true,
    rating: 3.1,
    countOfBedrooms: `3 Bedrooms`,
    maxCountOfGuests: `Max 4 adults`,
    equipment: [
      `Wifi`, `Heating`, `Kitchen`, `Cable TV`
    ],
    owner: {
      picture: `https://picsum.photos/74/74`,
      name: `Mike`,
      isSuper: true,
    },
    coords: [52.3909553943508, 4.85309666406198],
    reviews: [
      {
        name: `Max`,
        image: `https://picsum.photos/54/54`,
        rating: 4.1,
        description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
        date: `2019-04-24`
      }
    ]
  }];

it(`Если приложение загрузилось, то компонент PlaceCardDetails отрисовался`, () => {
  const tree = renderer
    .create(<PlaceCardDetails
      offer={mock}
      neighbourhoods={mockNeighbourhoods}
      onOfferTitleClick={() => {}}
    />, {
      createNodeMock: () => document.createElement(`div`)
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
