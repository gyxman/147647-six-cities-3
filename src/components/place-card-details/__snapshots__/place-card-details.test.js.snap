// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Если приложение загрузилось, то компонент PlaceCardDetails отрисовался 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Amazing and Extremely Central Flat"
            className="property__image"
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Amazing and Extremely Central Flat
          </h1>
          <button
            className="property__bookmark-button button"
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "40%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            2.2
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            4
          </li>
          <li
            className="property__feature property__feature--adults"
          />
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            348
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Breakfast
            </li>
            <li
              className="property__inside-item"
            >
              Washer
            </li>
            <li
              className="property__inside-item"
            >
              Laptop friendly workspace
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
            >
              <img
                alt="Angelina"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
          </div>
          <div
            className="property__description"
          >
            I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
    </section>
  </div>
</main>
`;
